src/runtime_src/tools/xclbinutil/KernelUtilities.cxx

```diff
@@ -129,7 +129,15 @@ void buildXMLKernelEntry(const boost::property_tree::ptree& ptKernel,
   const boost::property_tree::ptree& ptExtendedData = ptKernel.get_child("extended-data", ptEmpty);
   if (!ptExtendedData.empty()) {
     boost::property_tree::ptree ptEntry;
-    ptEntry.add_child("<xmlattr>", ptExtendedData);
+    boost::property_tree::ptree ptNewExtendedData;
+    const std::string& kernel_id = ptExtendedData.get<std::string>("dpu_kernel_id", "");
```

Comment:
Why the filter as opposed to using the ptExtendedData directly?  

The idea here was to take the "extended-data" from the input JSON file and added it directly to the XML's extended data.   By doing so there isn't a need to have this functionality.

Side note, if there is a need to support a given input schema, please do let me know and a rapidSchema validator can be added.

Reply:
We have discussed and we don't like the name dpu_kernel_id while already has type="dpu" in somewhere.

So are we OK with dpu_kernel_id here?
<kernel name="PP_FD_PRE" language="c" type="dpu">
         <extended-data functional="1“ kernel_id ="0x100">